src/cmd/run.js

```diff
@@ -42,16 +44,28 @@ export function defaultWatcherCreator(
   {
     addonId, client, sourceDir, artifactsDir,
     onSourceChange = defaultSourceWatcher,
+    desktopNotifications = notifier,
   }: WatcherCreatorParams
  ): Watchpack {
   return onSourceChange({
     sourceDir,
     artifactsDir,
     onChange: () => {
       log.debug(`Reloading add-on ID ${addonId}`);
+
       return client.reloadAddon(addonId)
+        .then(() => {
+          desktopNotifications({
```

Comment:
Here are some other ideas to keep this patch simple so it's easier to land:
* Only show a reload notification in verbose mode
* Remove the reload notification entirely and add it in later once we have a command line option